### DISCUSSION TITLE
Why is `@tanstack/react-query` a peer dependency?

### DISCUSSION THREAD

**Discussion by klippx** [trpc/trpc#4936] [Q&A] [answered]
In the docs it states:
> tRPC offers a first class integration with React. Under the hood this is simply a wrapper around the very popular [@tanstack/react-query](https://tanstack.com/query/latest), so we recommend that you familiarise yourself with React Query, as their docs go in to much greater depth on its usage.

[Source](https://trpc.io/docs/client/react)

Let me zoom in on that:
> a wrapper around [@tanstack/react-query](https://tanstack.com/query/latest)

If TRPC is a wrapper around another library, then that other library must be a STAPLE dependency for TRPC to work. If so, why do you want this to be a peer dependency? What flexibility does that provide?

I am using a Next app, so for me [these docs](https://trpc.io/docs/client/nextjs) is more relevant, but again:
> Our Next.js integration is built on top of our [React Query Integration](https://trpc.io/docs/)

We are back to being a wrapper around React Query. 

The problem I have is that now I have `@tanstack/react-query` in my dependencies, and I don't use it for anything other than satisfying trpc requirements. But I am not free to upgrade to v5, because that will break you (maybe). Wouldn't it be straight up a better experience for everyone if you specified `@tanstack/react-query` as a dependency instead, and let users that wants to mess with it (advanced case) use `resolutions` ?

**Comment by sachinraja** [comment]
There are definitely many use cases for using react query directly even if you are using tRPC. We have specified the version of react query that we support in peerDependencies.

**Reply by klippx** [reply]
Ok, sure, I don't disagree with that statement. So this is only for "reference", I don't actually _need_ to add `@tanstack/react-query` to my dependencies to satisfy this (soft) requirement? 💡

**Reply by Nick-Lucas** [reply, marked as answer]
It's not an optional peer dependency if you want to use the Tanstack Query integration

The purpose of peer deps is that you can install whatever version you like within the range we support, and this avoids issues like 2 different versions getting bundled up in your app and conflicting with each other. This would otherwise be possible if you're using it for your own needs, but we've also installed a version.

So us specifiying it as a dependency could actually cause you pain, whereas having to install it yourself isn't even really an inconvenience 😄

**Reply by klippx** [reply]
Thanks! I got confused because of in the example repo of "a minimal tRPC app" this repo actually had it installed as a dependency, but I am not sure why (I don't recall them using the Tanstack Query integration actually). So I have been feeling like a deer on ice trying to remove `@tanstack/react-query` despite it being part of minimal repo AND specified as peer dependency, but my app actually seem to work without it, so it all makes 100% sense now.
